### PR TITLE
Extend get cases filters

### PR DESCRIPTION
**Pull request by Sairef**
This pull request significantly expands the filtering capabilities for retrieving test cases from TestRail by adding support for multiple optional parameters. The changes update the API schema, server-side handler, and add comprehensive tests to ensure all new filters work as expected.

### API and Schema Enhancements

* Extended the `getCases` API endpoint to accept additional optional parameters for filtering: `createdBy`, `filter`, `milestoneId`, `priorityId`, `refs`, `sectionId`, `templateId`, `typeId`, `updatedBy`, and `labelId`, in addition to the existing `limit` and `offset`. This enables more precise queries for test cases.
* Updated the `getTestCasesSchema` in `cases.ts` to include validation for all new optional parameters, with detailed descriptions for each, ensuring robust and clear API documentation.

### Codebase Simplification

* Removed the direct dependency on `zod` in the server API handler, consolidating all schema validation through the shared schema for consistency.

### Test Coverage Improvements

* Added new tests in `cases.spec.ts` to verify that the API correctly handles all new optional parameters, including combinations and edge cases, ensuring reliability and backward compatibility.

### CODE REVIEW

**Comment by Sairef** [comment]
I just figured out that I was not able to find the tests by name, not only by ID. My use case was to ask Claude: 'Here is a folder with MD files, push the cases that do not exist and update the existing ones, filter by caseName listed in MD files', i suppose this PR will enable this workflow

**Comment by bun913** [comment]
@Sairef 

I was wondering if you could let me know what you want.
It looks like you want to filter by some fields.

But your changes look like:
- Accept custom parameters you add
- But it doesn't affect the expected result
  - I use TestRail API to fetch test case information
  - But the API doesn't have the filters like you want( in my understanding )
  - So it doesn't affect the expected result

> :filter return test cases which include the strings you added

https://docs.testrail.techmatrix.jp/testrail/docs/702/api/reference/cases/

File: src/server/api/cases.ts [hunk 1]
```diff
@@ -92,35 +91,62 @@ export function registerCaseTools(
 	// Get all test cases for a project
 	server.tool(
 		"getCases",
-		"Retrieves test cases list with basic fields only (excludes steps/expected results for performance). REQUIRED: projectId, suiteId. OPTIONAL: limit (default 50), offset (default 0). Use getCase for full details.",
+		"Retrieves test cases list with basic fields only (excludes steps/expected results for performance). REQUIRED: projectId, suiteId. OPTIONAL: createdBy, filter, limit (default 50), milestoneId, offset (default 0), priorityId, refs, sectionId, templateId, typeId, updatedBy, labelId. Use getCase for full details.",
```
Comment (bun913) [inline src/server/api/cases.ts]: What an informative information is!

File: src/server/api/cases.ts [hunk 2]
```diff
@@ -92,35 +91,62 @@ export function registerCaseTools(
...
-					"Number of cases to return per page. If you cannot get all cases, try separating the request into multiple calls",
-				),
-			offset: z
-				.number()
-				.optional()
-				.default(0)
-				.describe("Offset for pagination"),
+			createdBy: getTestCasesSchema.shape.createdBy,
+			filter: getTestCasesSchema.shape.filter,
+			limit: getTestCasesSchema.shape.limit,
```
Comment (bun913) [inline src/server/api/cases.ts]: Thank you for nice changing!

File: test/client/api/cases.spec.ts [hunk 3]
```diff
@@ -199,6 +199,185 @@ describe('Cases API', () => {
...
+        suite_id: 1,
+        filter: 'Registration',
+        refs: 'TR-456',
+        limit: 50,
+        offset: 0
+      }
+    });
+
+    // Verify result
+    expect(result).toEqual(mockCasesResponse);
```
Comment (bun913) [inline test/client/api/cases.spec.ts]:
```suggestion
```
It's same as the above one 🙇‍♂️

File: test/client/api/cases.spec.ts [hunk 4]
```diff
@@ -199,6 +199,185 @@ describe('Cases API', () => {
...
+      params: {
+        suite_id: 1,
+        section_id: 5,
+        limit: 50,
+        offset: 0
+      }
+    });
+
+    // Verify result
+    expect(result).toEqual(mockCasesResponse);
```
Comment (bun913) [inline test/client/api/cases.spec.ts]:
```suggestion
```

It's also same as the above one! 🙇

**Comment by Sairef** [comment]
@bun913 Yes, i see your point here. Updated the tests just to check that we do proper requests
